sites/hashdev/src/components/BlogPost.tsx

```diff
@@ -9,6 +8,36 @@ import { Link } from "./Link";
 import { mdxImageClasses } from "./MdxImage";
 import { FaIcon } from "./icons/FaIcon";
 
+/**
+ * @param {number} digit
+ * @returns the ordinal suffix of the input digit
+ */
+const nth = (digit: number) => {
+  if (digit > 3 && digit < 21) return "th";
+  switch (digit % 10) {
+    case 1:
+      return "st";
+    case 2:
+      return "nd";
+    case 3:
+      return "rd";
+    default:
+      return "th";
+  }
+};
+
+/**
+ * @param {Date} date
+ * @returns the formatted string of the date object (e.g. "May 17th, 2022")
+ */
+const formatDate = (date: Date) => {
+  const year = date.toLocaleString("default", { year: "numeric" });
+  const month = date.toLocaleString("default", { month: "long" });
+  const day = parseInt(date.toLocaleString("default", { day: "numeric" }), 10);
+
+  return `${month} ${day}${nth(day)}, ${year}`;
```

Comment:
Good idea, thanks Ciaran have made use of `date-fns` in https://github.com/hashintel/hash/pull/575/commits/44f1c85db37fd8dfd0fa53c68bb614a36ce53809. Have stuck with the default locale for now instead of using `window.navigator.languages[0]`.